File: sample/src/main/java/contacts/sample/cheatsheet/basics/java/QueryContactsByPhoneOrSipActivity.java
package contacts.sample.cheatsheet.basics.java;

import android.app.Activity;

import java.util.List;

import contacts.core.ContactsFactory;
import contacts.core.PhoneLookupQuery;
import contacts.core.entities.Contact;

public class QueryContactsByPhoneOrSipActivity extends Activity {

    List<Contact> getContactsWithPhoneNumberThatExactlyMatches(String text) {
        return ContactsFactory.create(this)
                .phoneLookupQuery()
                .match(PhoneLookupQuery.Match.PHONE)
                .whereExactlyMatches(text)
                .find();
    }

    List<Contact> getContactsWithEmailOrDisplayNameThatPartiallyMatches(String text) {
        return ContactsFactory.create(this)
                .phoneLookupQuery()
                .match(PhoneLookupQuery.Match.SIP)
                .whereExactlyMatches(text)
                .find();
    }
}

File: sample/src/main/java/contacts/sample/cheatsheet/basics/kotlin/QueryContactsByPhoneOrSipActivity.kt
package contacts.sample.cheatsheet.basics.kotlin

import android.app.Activity
import contacts.core.*
import contacts.core.entities.Contact

class QueryContactsByPhoneOrSipActivity : Activity() {

    fun getContactsWithPhoneNumberThatExactlyMatches(text: String?): List<Contact> =
        Contacts(this)
            .phoneLookupQuery()
            .match(PhoneLookupQuery.Match.PHONE_NUMBER)
            .whereExactlyMatches(text)
            .find()

    fun getContactsWithSipAddressThatExactlyMatches(text: String?): List<Contact> =
        Contacts(this)
            .phoneLookupQuery()
            .match(PhoneLookupQuery.Match.SIP_ADDRESS)
            .whereExactlyMatches(text)
            .find()
}